dgraph/cmd/graphql/schema/schema_test.go

```diff
@@ -134,56 +82,65 @@ func TestEqualSchema(t *testing.T) {
 		str1, err := ioutil.ReadFile(fileName1)
 		if err != nil {
 			t.Errorf("Unable to read file %s", fileName1)
-		}
-
-		doc1, gqlerr := parser.ParseSchema(&ast.Source{Input: string(str1)})
-		if gqlerr != nil {
-			t.Errorf("Unable to parse file %s\n%s", fileName1, gqlerr.Error())
 			continue
 		}
 
-		gqlErrList1 := ValidateSchema(doc1)
-		if gqlErrList1 != nil {
-			t.Errorf("Unable to validate schemafile %s\n%s", fileName1, gqlErrList1.Error())
-		}
-
-		AddScalars(doc1)
-
-		schema1, gqlerr := validator.ValidateSchemaDocument(doc1)
-		if gqlerr != nil {
-			t.Errorf("Unable to validate schemafile %s\n%s", fileName1, gqlerr.Error())
+		sch1, errlist1 := GenerateCompleteSchema(string(str1))
+		if errlist1 != nil {
+			t.Errorf(errlist1.Error())
 			continue
 		}
 
-		GenerateCompleteSchema(schema1)
-
 		fileName2 := "../testdata/equalschema" + strconv.Itoa(i) + "_2.txt" // run from pwd
 		str2, err := ioutil.ReadFile(fileName2)
 		if err != nil {
 			t.Errorf("Unable to read file %s", fileName2)
 		}
 
-		doc2, gqlerr := parser.ParseSchema(&ast.Source{Input: string(str2)})
-		if gqlerr != nil {
-			t.Errorf("Unable to parse file %s\n%s", fileName2, gqlerr.Error())
+		sch2, errlist2 := GenerateCompleteSchema(string(str2))
+		if errlist2 != nil {
+			t.Errorf(errlist2.Error())
 			continue
 		}
 
-		gqlErrList2 := ValidateSchema(doc2)
-		if gqlErrList2 != nil {
-			t.Errorf("Unable to validate schemafile %s\n%s", fileName2, gqlErrList2.Error())
-		}
+		require.True(t, AreEqualSchema(sch1, sch2))
+	}
+}
 
-		AddScalars(doc2)
+type Tests map[string][]TestCase
+type TestCase struct {
+	Name   string
+	Input  string
+	Output gqlerror.List
+}
 
-		schema2, gqlerr := validator.ValidateSchemaDocument(doc2)
-		if gqlerr != nil {
-			t.Errorf("Unable to validate schemafile %s\n%s", fileName2, gqlerr.Error())
-			continue
-		}
+func TestInvalidSchemas(t *testing.T) {
+	fileName := "schema_test.yml" // run from pwd
```

Comment:
So these tests won't work if executed from the root of the repo? You could get the path of the file using `$GOPATH`.